src/utils/analytics/events.ts

```diff
@@ -0,0 +1,169 @@
+import {track} from 'utils/analytics'
+
+//! ENGAGEMENT EVENT GROUP
+// start course
+/* When a course get started by a user  */
+/* This will show courses that get started but the don't get finished  */
+
+export const engagementStartCourse = (slug: string) =>
+  track('start course', {
+    slug,
+  })
```

Comment:
included the trigger and event description for each event, this would be nice to add to the typescript overlay but haven't looked into that.